source/guides/continuous-integration/aws-codebuild.md

```diff
@@ -0,0 +1,247 @@
+```
+
+The `build-matrix` will run all permutations delimited items.  
+
+```yaml
+batch:
+  fast-fail: false
+  build-matrix:
+    # ...
+    dynamic:
+      env:
+        # ...
+        variables:
+          CY_GROUP_SPEC:
+            - "UI - Chrome|chrome|cypress/tests/ui/*"
+            - "UI - Chrome - Mobile|chrome|cypress/tests/ui/*|viewportWidth=375,viewportHeight=667"
+            - "API|chrome|cypress/tests/api/*"
+            - "UI - Firefox|firefox|cypress/tests/ui/*"
+            - "UI - Firefox - Mobile|firefox|cypress/tests/ui/*|viewportWidth=375,viewportHeight=667"
+
+```
+
+During the install phase, we utilize shell scripting with the {% url "cut command" https://en.wikipedia.org/wiki/Cut_(Unix) %} to assign values from the delimited `CY_GROUP_SPEC` passed to the worker into shell variables that will be used in the `build` phase when running `cypress run`.
+
+```yaml
+batch:
+  # ...
+
+phases:
+  install:
+    commands:
+      - CY_GROUP=$(echo $CY_GROUP_SPEC | cut -d'|' -f1)
+      - CY_BROWSER=$(echo $CY_GROUP_SPEC | cut -d'|' -f2)
+      - CY_SPEC=$(echo $CY_GROUP_SPEC | cut -d'|' -f3)
+      - CY_CONFIG=$(echo $CY_GROUP_SPEC | cut -d'|' -f4)
+      - npm ci
+# ...
+```
+
+To parallelize the runs, we need to add an additional variable to the {% url "build-matrix strategy" https://docs.aws.amazon.com/codebuild/latest/userguide/batch-build-buildspec.html#build-spec.batch.build-matrix %}, `WORKERS`.
+
+```yaml
+batch:
+  fast-fail: false
+  build-matrix:
+    # ...
+    dynamic:
+      env:
+        # ...
+        variables:
+          CY_GROUP_SPEC:
+            # ...
+          WORKERS:
+            - 1
+            - 2
+            - 3
+            - 4
+            - 5
+```
+{% note info Note %}
+The `WORKERS` array is filled with filler (or *dummy*) items to provision the desired number of CI machine instances within the {% url "build-matrix strategy" https://docs.aws.amazon.com/codebuild/latest/userguide/batch-build-buildspec.html#build-spec.batch.build-matrix %} and will provide 5 workers to each group defined in the `CY_GROUP_SPEC`.
+{% endnote %}
+
+Finally, the script variables are passed to the call to `cypress run`.
+
+```yaml
+phases:
+  install:
+    # ...
+  build:
+    commands:
+      - npm start:ci &
+      - npx cypress run --record --parallel --browser $CY_BROWSER --ci-build-id $CODEBUILD_INITIATOR --group "$CY_GROUP" --spec "$CY_SPEC" --config "$CY_CONFIG"
+```
+
+{% note info %}
+
+# Using the Cypress Dashboard with AWS CodeBuild
+Finally, we tell Cypress command to record results to the {% url "Cypress Dashboard" https://on.cypress.io/dashboard %} (using the `CYPRESS_RECORD_KEY` environment variable) in parallel.
+
+Jobs can be organized by groups and in this job we specify a `--group "$CY_GROUP"` to consolidate all runs for these workers in a central location in the {% url "Cypress Dashboard" https://on.cypress.io/dashboard %}.
```

Comment:
@kevinold let's have this section reach some parity with the other CI guides.

Reply:
Updated